### DISCUSSION TITLE
[TOPAS validation]: Anisotropy function for Flexisource 192Ir

### DISCUSSION THREAD

**Discussion by kh-434** [OpenTOPAS/OpenTOPAS#70] [Validation]
### Describe your simulation result

Hello everyone! 

I am new to TOPAS, I am using Topas Version 3.9. I am trying to replicate the Anisotropy function for Flexisource 192Ir  from D. Granero, 2006. and I keep getting weird results after normalizing the data of 10mm to the 90-degree angle and plotting them with reference data and it isn't clear. I tried different production cuts, different step sizes, different physics, and different voxel sizes, and nothing is working. 

I have shared the full code and results. 

Best regards, 
[Flexisource Ir-192.txt](https://github.com/user-attachments/files/16667019/Flexisource.Ir-192.txt)

<img width="1063" alt="Screenshot 2024-08-20 at 10 02 57 AM" src="https://github.com/user-attachments/assets/a4fc6fc6-076a-470f-a61f-29d898cf99ef">

### Version

3.9

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment]
Hi there,
Please use the updated version v4.0  https://github.com/OpenTOPAS/OpenTOPAS.git
Best.

**Comment by jschuemann** [comment]
Hi,
just to add to this, we have moved to a new version. This is a major upgrade that fixes many small things and uses a new Geant4 version, which also has many new features and updates. We are a small team, and we have to limit answering on the forum to the latest version as many problems with prior versions may already be fixed.
Thank you for reporting your question, please let us know if you still have the same problem in the new version, OpenTOPAS 4.0.

**Comment by kh-434** [comment]
Hi,

Thank you for your suggestion. I have updated it to OpenTOPAS 4.0. But I got the same results; any help will be highly appreciated as I have been struggling with the same issue for weeks. 

Kind regards,

**Comment by JoseRamosMendez** [comment]
Hi there,
The scorer geometry overlaps with the source. There must no overlaps of mass geometries in a Monte Carlo simulation. Given your geometry setup, you can fix this by setting the AnisotropyPhantom **parallel** as follows:
     
     #sc:Ge/AnisotropyPhantom/Material = "G4_WATER"
     b:Ge/AnisotropyPhantom/IsParallel = "True"

Best!

**Reply by kh-434** [reply]
Hi Jose, 

Thank you for your response, 

I got these errors: 
Topas experienced an error in scoring.
G4WT1 > A scorer in the Component: "AnisotropyPhantom"
G4WT1 > has been called for a hit in the non-parameterized volume named: "AnisotropyPhantom"
G4WT1 > But this deos not make sense, as this scorer should only get called for hits
G4WT1 > in a parameterized volume (volume name would include the string "_Division").
G4WT1 > We think this may sometimes happen because Geant4 navigation gets into a corrupted state.
G4WT1 > This hit will be omitted from scoring.
G4WT1 > A note at end of session will show total number of unscored steps and the total unscored energy.
G4WT1 > In some cases the unscored energy is small enough that you can ignore this issue.
G4WT1 > This hit's energy deposit would have been: 1.9482e-06 MeV

Topas experienced an error in scoring.
G4WT1 > A scorer in the Component: "AnisotropyPhantom"
G4WT1 > has been called for a hit in the non-parameterized volume named: "AnisotropyPhantom"
G4WT1 > But this deos not make sense, as this scorer should only get called for hits
G4WT1 > in a parameterized volume (volume name would include the string "_Division").
G4WT1 > We think this may sometimes happen because Geant4 navigation gets into a corrupted state.
^CG4WT1 > This hit will be omitted from scoring.

How to solve it ? 

Best regards,

**Comment by JoseRamosMendez** [comment]
That warning is caused by the binned geometry. You can partially mitigate that issue by simplifying it, like using a single binned ring and trying to avoid overlaps, e.g., use parallel worlds only if absolutely needed.
